### ISSUE TITLE
Mediashout Lyrics

### ISSUE THREAD

**Issue by rohanale** [ChrisMBarr/LyricConverter#58]
Hi,

Stumbled upon your program which looks great.

Was wondering if it works with Mediashout v7 which we used at our church.  I believe they used json format for their lyrics.

I have a bunch of songs in text format which I would like to convert so they can be used in Mediashout if possible.

Thanks

Rohan

**Comment by FiniteLooper** [comment]
It does not currently work with MediaShout, but that's mostly because I have not ever had any sample files to test with. I cannot guarantee when, but if you have any MediaShout files you can send over I can take a look at the format to see if it's something I can add support for.

**Comment by rohanale** [comment]
Hi Chris, 

Thanks for getting back to me. 

I'll send you a sample file later. 

Thanks 

Rohan

**Comment by rohanale** [comment]
Hi Chris, 

Here is a sample file. 

Hope that helps. 

Thanks 

Rohan

**Comment by FiniteLooper** [comment]
I don't think you can reply to the github emails with attachments, they don't get passed through.  You'll have to do it on the github website

**Comment by rohanale** [comment]
Here you go ...
[JOY is Joy.json](https://github.com/FiniteLooper/LyricConverter/files/14466961/JOY.is.Joy.json)

**Comment by rohanale** [comment]
[The Blessing.json](https://github.com/FiniteLooper/LyricConverter/files/14467389/The.Blessing.json)

here's another song.

**Comment by rohanale** [comment]
[Hymn 002 - All Creatures or our God.json](https://github.com/FiniteLooper/LyricConverter/files/14467393/Hymn.002.-.All.Creatures.or.our.God.json)

**Comment by rohanale** [comment]
Hi. Any news on those mediashout files I sent?

**Comment by FiniteLooper** [comment]
Sorry, I haven't had time to look into this yet. I would need to analyze the files, learn about the format, and build an entire parsing/building library and then integrate it into LyricConverter. I will attempt to get to it when I have the time, but probably not something terribly soon.

**Comment by rohanale** [comment]
Hi Chris.

No worries.

Thanks

Rohan

**Comment by ChrisMBarr** [comment]
@rohanale I began to take a look at this, but I am unclear on a few things about these files.

What version of MediaShout are you using to get those `*.json` files?  I installed the trial version of Mediashout 7 and I see no options at all for `*.json` files. It seems to save things as `*.sc7x` files, which I would not be able to do anything with (they are binary files).

The JSON file you sent would be extremely easy to work with, but I am unable to open them in MediaShout and I am also unable to get MediaShout to export anything in that format.

**Comment by rohanale** [comment]
Hi Chris,

So I too am using Mediashout 7. The .scx files are the format for the scripts. 

When you go into Mediashout open one of the 2 sample scripts by double clicking on it. Any one is fine. Then just either click on the note icon in the top right or choose Insert Lyric Cue.

You should see a list of songs. All the lyrics/songs listed are saved in json format.You can export and import songs from here.

Hope that helps. Let me know if you have any more questions. 

Thanks.

Rohan

**Comment by ChrisMBarr** [comment]
ok, sorry, I've been busy. Finally sat down to try this out and I'm a little lost. I used MediaShout probably 15+ years ago and it has changed quite a lot!

You said _"All the lyrics/songs listed are saved in JSON format.You can export and import songs from here."_ but I don't see any options for that anywhere.  I do see some "export" options but my only options are text, PDF, or images. I'm not seeing JSON listed anywhere

![image](https://github.com/ChrisMBarr/LyricConverter/assets/463685/16ba7a7b-fab3-42a9-9ee5-067f57d17755)

![image](https://github.com/ChrisMBarr/LyricConverter/assets/463685/0b2974ad-ff6f-4944-9865-3cfd1ddaa7d9)

In addition to that, I'm not seeing any way to import these files anyway.  There's a very clear "Import" menu, but no JSON is listed.

![image](https://github.com/ChrisMBarr/LyricConverter/assets/463685/28d612fc-e67c-47ff-aa73-af12191b0d70)

But after looking in their knowledge base, I found this page: https://support.mediashout.com/244705-How-to-move-your-Song-Library-from-one-computer-to-another---MediaShout-7

Basically there are unlabeled icons in a menu, and I have no idea why they would do it this way and not list this in the regular import/export menus. Whatever!  I'm just glad their file formats are simple JSON files which are incredibly easy to work with.

![image](https://github.com/ChrisMBarr/LyricConverter/assets/463685/8fbb6e0b-cd2a-417d-b9fc-40c02cdf14ad)

**Comment by rohanale** [comment]
Hi Chris,

I will send you some further instructions of how to access the json files shortly.

Thanks.

Rohan  

⁣Get BlueMail for Android ​

**Comment by ChrisMBarr** [comment]
No that's OK, I've got it working. I was just venting about how un-obvious it is to get that.

I've actually currently got it working with reading/importing MediaShout files. Still need to work on the outputs.  MediaShould also allow multiple songs to be in a single file to be imported, so I will need to handle a way to deal with that as well. That will be useful for importing an entire library

**Comment by rohanale** [comment]
Oh wow. Great stuff. Let me know if you need anything from me.

**Comment by ChrisMBarr** [comment]
I've just published a new release which should be live on the website now.  LyricConverter can now READ MediaShout 7 files and convert them to any of the output types! This works for files that contain multiple songs too!

I wanted to get the MediaShout file output type made as well, but I ran out of time this weekend. But that's a separate feature really so I wanted to get this out there now in case it was needed.

Please let me know if this works for you or if you have any issues with it reading MS7 files.  I'm going to keep this issue open until I complete the MS7 Output type as well.

**Comment by rohanale** [comment]
Excellent Chris. I will give it a try and let you know. Thanks for your time and effort spent on this.

Regards

Rohan

**Comment by rohanale** [comment]
Hi Chris,

Just looked on your website and realise what you meant about the Output type.  So we had all the songs in Mediashout but due to a machine issue, lost all the songs.

So the output type is very important for me.  I need it as we have all the songs in text format which I would like to convert to Mediashout 7 json format so I do not have the type the songs back in again.

Thanks

Rohan

**Comment by ChrisMBarr** [comment]
ok, I've just finished up the MediaShout 7 support for output/writing files!  Please give this a try and let me know how this works for you.  I'm going to go ahead and close this issue for now, if you have any problems please open up a new issue

**Comment by rohanale** [comment]
Hi Chris,

Thanks for sorting this.  It is working well.

The only comment I would make is that when the song is created in Mediashout all the stanzas are labeled as Verse1.  So if the song has 2 verses for example both verses are labelled as Verse 1.  And choruses are not recognised.  I guess there is nothing in the text to denote what a verse or a chorus is.  

Not sure if anything can be done about this but I thought I would let you know.

Thanks

Rohan

**Comment by ChrisMBarr** [comment]
Hmm strange. I might be able to do something about that. Can you please create a new issue here and attach a file.

If you're curious, a mediashout file just uses a number to define which part of a song it is, like this: https://github.com/ChrisMBarr/LyricConverter/blob/master/test/sample-files/MediaShout/JOY.is.Joy.json#L13

And then I have a way to map those numbers to which part they should be here: https://github.com/ChrisMBarr/LyricConverter/blob/master/src/app/convert/models/mediashout.model.ts#L29-L45

I just figured that out based on the dropdown list inside of MediaShout, it appears to be in the correct order.
